### PR TITLE
Add more functional tests for the search service

### PR DESCRIPTION
**Pull request by joelverhagen**
Per NuGet/NuGetGallery#2798.

1. Download count test for V3 search.
2. Empty result test for all three endpoints.
3. Schema test for V3 search.
4. Prerelease and order by downloads test for V2 search.
4. Fix web.config so `searchcloud` branch can deploy to Azure.

@johnataylor @maartenba @xavierdecoster @joyhui

### CODE REVIEW

File: src/NuGet.Services.BasicSearch/Web.config [hunk 1]
```diff
@@ -13,12 +13,12 @@
     <customErrors mode="Off" />
   </system.web>
   <appSettings>
+    <!--
     <add key="Local.Lucene.Directory" value="C:\data\index20151120" />
```
Comment (joyhui) [inline src/NuGet.Services.BasicSearch/Web.config]: <add key="Local.Lucene.Directory" value="{local path. e.g. C:\data\index}" />

File: tests/NuGet.Services.BasicSearchTests/TestSupport/NupkgDownloader.cs [hunk 2]
```diff
@@ -53,24 +56,34 @@ public string GetPackagePath(PackageVersion version)
...
-            if (_packageBaseAddress == null)
-            {
-                _packageBaseAddress = await GetPackageBaseAddressAsync();
-            }
-            
-            string relativeUri = $"{version.Id}/{version.Version}/{version.Id}.{version.Version}.nupkg".ToLower();
-            var requestUri = new Uri(new Uri(_packageBaseAddress, UriKind.Absolute), relativeUri);
-            var response = await _client.GetAsync(requestUri);
-            if (response.StatusCode != HttpStatusCode.OK)
+            var acquired = await PackageLock.WaitAsync(TimeSpan.FromSeconds(30));
```
Comment (joyhui) [inline tests/NuGet.Services.BasicSearchTests/TestSupport/NupkgDownloader.cs]: move acquire step before write file to make them real parallel.

File: tests/NuGet.Services.BasicSearchTests/TestSupport/StartedWebApp.cs [hunk 3]
```diff
@@ -56,10 +57,10 @@ private StartedWebApp()
             // set up the data directory
             var loader = new InMemoryLoader
             {
-                { "downloads.v1.json", "[]" },
+                { "downloads.v1.json", BuildDownloadsFile(enumeratedPackages) },
                 { "curatedfeeds.json", "[]" },
                 { "owners.json", "[]" },
-                { "rankings.v1.json", "{\"Rank\": []}" }
+                { "rankings.v1.json", BuildRankingsFIle(enumeratedPackages) }
```
Comment (maartenba) [inline tests/NuGet.Services.BasicSearchTests/TestSupport/StartedWebApp.cs]: Nit: capital `I` -> `i`

**Comment by maartenba** [comment]
Apart from nit: :shipit:

**Comment by xavierdecoster** [comment]
:shipit: when comments addressed

File: src/NuGet.Services.BasicSearch/Web.config (hunk 1 above)
Comment (joelverhagen) [inline src/NuGet.Services.BasicSearch/Web.config]: Fixed.

File: tests/NuGet.Services.BasicSearchTests/TestSupport/StartedWebApp.cs (hunk 3 above)
Comment (joelverhagen) [inline tests/NuGet.Services.BasicSearchTests/TestSupport/StartedWebApp.cs]: Fixed.

File: tests/NuGet.Services.BasicSearchTests/TestSupport/NupkgDownloader.cs (hunk 2 above)
Comment (joelverhagen) [inline tests/NuGet.Services.BasicSearchTests/TestSupport/NupkgDownloader.cs]: Fixed.

**Pull request merged**